app/src/js/controllers/WorkspaceController.js

```diff
@@ -143,11 +136,19 @@ angular.module('media_manager')
       self.isLoadingCollection.status = true;
       collection.$promise.then(function(collection) {
         wc.collection = collection;
+        
+        // update the CourseCache.collections since it is stale
+        var collection_idx = wc.courseCollections.map(function(c) { return c.id; }).indexOf(collection.id);
+        if (collection_idx >= 0) {
+          wc.courseCollections.splice(collection_idx, 1, collection);
+        }
+        $log.debug("insert updated collection into cache at index: ", collection_idx);
```

Comment:
The code above is what fixes the issue. The CourseCache is not directly updated when the collection is saved because `wc.collection` refers to an object *outside* of the cache.